velox/type/tests/VariantTest.cpp

```diff
@@ -89,3 +89,57 @@ TEST(Variant, opaque) {
     EXPECT_NE(v1, vint);
   }
 }
+
+TEST(Variant, shortDecimal) {
+  auto shortDecimalType = DECIMAL(10, 3);
+  variant v = variant::shortDecimal(1234, shortDecimalType);
+  EXPECT_TRUE(v.hasValue());
+  EXPECT_EQ(TypeKind::SHORT_DECIMAL, v.kind());
+  EXPECT_EQ(1234, v.value<TypeKind::SHORT_DECIMAL>().value().unscaledValue());
+  EXPECT_EQ(10, v.value<TypeKind::SHORT_DECIMAL>().precision);
+  EXPECT_EQ(3, v.value<TypeKind::SHORT_DECIMAL>().scale);
+  EXPECT_EQ(*v.inferType(), *shortDecimalType);
+  EXPECT_EQ(v.toJson(), "1.234");
+  // 1.2345
+  variant u1 = variant::shortDecimal(12345, DECIMAL(10, 4));
+  // 1.2345 > 1.234
+  EXPECT_LT(
+      v.value<TypeKind::SHORT_DECIMAL>(), u1.value<TypeKind::SHORT_DECIMAL>());
+  // 0.1234
+  variant u2 = variant::shortDecimal(1234, DECIMAL(10, 4));
+  // 0.1234 < 1.234
+  EXPECT_LT(
+      u2.value<TypeKind::SHORT_DECIMAL>(), v.value<TypeKind::SHORT_DECIMAL>());
+  variant null = variant::shortDecimal(std::nullopt, DECIMAL(10, 5));
```

Comment:
nit: perhaps, extract this logic into a separate test or separate it with a comment for readability

Reply:
moved the logic as separate tests